view/Color.cpp

```diff
@@ -0,0 +1,36 @@
+#include "view/Color.h"
+
+short ncurses_color(Color color)
+{
+	switch (color)
+	{
+		case Color::Black:
+			return COLOR_BLACK;
+			break;
+		case Color::Red:
+			return COLOR_RED;
+			break;
+		case Color::Green:
+			return COLOR_GREEN;
+			break;
+		case Color::Yellow:
+			return COLOR_YELLOW;
+			break;
+		case Color::Blue:
+			return COLOR_BLUE;
+			break;
+		case Color::Magenta:
+			return COLOR_MAGENTA;
+			break;
+		case Color::Cyan:
+			return COLOR_CYAN;
+			break;
+		case Color::White:
+			return COLOR_WHITE;
+			break;
+		default:
+			throw std::logic_error("In file Color.cpp on line 33:\n    unexpected color value in case section\n");
```

Comment:
Замучаешься поддерживать:) Небольшое изменение в название файла будет приводить к дезактуализации строки.

Попробуй завести вспомогательный макрос: 
`#define MSG_ERR "In file " __FILE__ " on line " __LINE__ ":"`

Можно будет заменить текущую строку на:
```
throw std::logic_error(MSG_ERR "\n    unexpected color value in case section\n");

```

Reply:
Я хотел так сделать, но `__LINE__ ` - это int, поэтому его нельзя просто так в строку впихнуть. Если можно, скажи как.

Reply:
С макросом `LINE_STRING` годный вариант.

Reply:
Done: 913252280b337fbbd4694cfe165f01e46d2819b1

Reply:
`+`